src/node/desktop/src/renderer/desktop-bridge.ts

```diff
@@ -69,20 +68,28 @@ export function getDesktopBridge() {
       ipcRenderer
         .invoke('desktop_get_save_file_name', caption, label, dir, defaultExtension, forceDefaultExtension, focusOwner)
         .then((result) => {
+
+          // if the result was canceled, bail early
           if (result.canceled as boolean) {
-            callback('');
-          } else {
-            if (defaultExtension.length !== 0) {
-              // Add default extension, if it's missing
-              const fp = new FilePath(result.filePath);
-              if (fp.getExtension().length == 0 || (forceDefaultExtension && fp.getExtension() !== defaultExtension)) {
-                callback(fp.getStem() + defaultExtension);
-                return;
-              }
-            }
-
-            callback(result.filePath);
+            return callback('');
+          }
+          
+          // if we don't have a default extension, just invoke callback
+          let filePath = result.filePath as string;
+          if (defaultExtension.length === 0) {
+            return callback(filePath);
           }
+
+          // add default extension if necessary
+          const dotIndex = filePath.lastIndexOf('.');
+          const ext = dotIndex > 0 ? filePath.substring(dotIndex) : '';
```

Comment:
Note that we explicitly check for `dotIndex > 0` here to handle dotfiles, e.g. `.Rprofile`.